tasks/html.js

```diff
@@ -8,41 +8,62 @@
 
 'use strict';
 
+var path = require('path');
 var htmllint = require('../lib/htmllint');
+var reporters = require('../lib/reporters');
 
 module.exports = function(grunt) {
 
-  var chalk = require('chalk');
-
   grunt.registerMultiTask('htmllint', 'Validate html files', function() {
     var done = this.async(),
       files = grunt.file.expand(this.filesSrc),
       options = this.options({
         files: files,
-        force: false
+        force: false,
+        absoluteFilePathsForReporter: false
       }),
-      force = options.force;
+      force = options.force,
+      reporterOutput = options.reporterOutput,
+      reporter;
 
     htmllint(options, function(error, result) {
+      var passed = true,
+        output;
+
+      try {
+        reporter = reporters.selectReporter(options);
+      } catch (err) {
+        grunt.fatal(err);
+      }
+
       if (error) {
+        passed = force;
         grunt.log.error(error);
-        done(force);
-        return;
       }
-      if (!result.length) {
-        grunt.log.ok(files.length + ' file' + (files.length === 1 ? '' : 's') + ' lint free.');
-        done();
-        return;
+      else if (!result.length) {
+        grunt.log.ok(files.length + ' ' + grunt.util.pluralize(files.length, 'file/files') + ' lint free.');
       } else {
-        result.forEach(function(message) {
-          var output = chalk.cyan(message.file) + ' ';
-          output += chalk.red('[') + chalk.yellow('L' + message.lastLine) +
-            chalk.red(':') + chalk.yellow('C' + message.lastColumn) + chalk.red('] ');
-          output += message.message;
+        passed = force;
+        output = reporter(result);
+        if (!reporterOutput) {
           grunt.log.writeln(output);
-        });
+        }
+        grunt.log.error(result.length + ' ' + grunt.util.pluralize(result.length, 'error/errors') + ' in ' +
+                        files.length + ' ' + grunt.util.pluralize(files.length, 'file/files'));
```

Comment:
This is a bit odd. Running `grunt htmllint:invalid` it outputs ">> 4 errors in 2 files", while the 4 errors are in just one file. Might be better  to only output the number of errors, or be more explicit about how many files were validated and how many of those had errors.